doc/api.md

```diff
@@ -65,6 +81,99 @@ var api = new JitsiMeetExternalAPI(domain, options);
 
 ### Controlling the embedded Jitsi Meet Conference
 
+You can control the available devices  with the following methods of `JitsiMeetExternalAPI` instance:
+* **getAvailableDevices** - Retrieve a list of available devices.
+
+```javascript
+api.getAvailableDevices().then(function(devices) {
+    // devices = {
+    //     'audioInput': [{
+    //         deviceId: "ID"
+    //         groupId: "grpID"
+    //         kind: "audioinput"
+    //         label: "Label"
+    //     },....],
+    //     'audioOutput': [{
+    //         deviceId: "ID"
+    //         groupId: "grpID"
+    //         kind: "audioOutput"
+    //         label: "Label"
+    //     },....],
+    //     'videoInput': [{
+    //         deviceId: "ID"
+    //         groupId: "grpID"
+    //         kind: "videoInput"
+    //         label: "Label"
+    //     },....]
+    // }
+    ...
+});
+```
+* **getCurrentDevices** - Retrieve a list with the devices that are currently selected.
+
+```javascript
+api.getCurrentDevices().then(function(devices) {
+    // devices = {
+    //     'audioInput': {
+    //         deviceId: "ID"
+    //         groupId: "grpID"
+    //         kind: "videoInput"
+    //         label: "Label"
+    //     },
+    //     'audioOutput': {
+    //         deviceId: "ID"
+    //         groupId: "grpID"
+    //         kind: "videoInput"
+    //         label: "Label"
+    //     },
+    //     'videoInput': {
+    //         deviceId: "ID"
+    //         groupId: "grpID"
+    //         kind: "videoInput"
+    //         label: "Label"
+    //     }
+    // }
+    ...
+});
+```
+* **isDeviceChangeAvailable** - Resolves with true if the device change is available and with false if not.
+
+```javascript
+// The accepted deviceType values are - 'output', 'input' or undefined.
+api.isDeviceChangeAvailable(deviceType).then(function(isDeviceChangeAvailable) {
+    ...
+});
+```
+* **isDeviceListAvailable** - Resolves with true if the device list is available and with false if not.
+
+```javascript
+api.isDeviceListAvailable().then(function(isDeviceListAvailable) {
+    ...
+});
+```
+* **isMultipleAudioInputSupported** - Resolves with true if the device list is available and with false if not.
+
+```javascript
+api.isMultipleAudioInputSupported().then(function(isMultipleAudioInputSupported) {
+    ...
+});
+```
+* **setAudioInputDevice** - Sets the audio input device to the one with the label or id that is passed.
+
+```javascript
+api.setAudioInputDevice(deviceLabel, deviceId);
+```
+* **setAudioOutputDevice** - Sets the audio output device to the one with the label or id that is passed.
+
+```javascript
+api.setAudioOutputDevice(deviceLabel, deviceId);
+```
+* **setVideoInputDevice** - Sets the video input device to the one with the label or id that is passed.
+
+```javascript
+api.setVideoInputDevice(deviceLabel, deviceId);
```

Comment:
Is it actually useful to support id and label? I would think id is generally a valid case for the device selection popup but otherwise label is the valid case.

Reply:
I decided to document the id support because I had to implement it for the device selection popup. Maybe somebody will have a weird use case where the deviceId will be useful...